### PR TITLE
bpo-40778: Update Py_REFCNT and Py_SIZE to const casting

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue40778

### CODE REVIEW

File: Misc/NEWS.d/next/C API/2020-05-26-23-06-59.bpo-40778.7cumE-.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Update :c:macro:`Py_SIZE` and :c:macro:`Py_SIZE` to use const object
+casting. Patch by Dong-hee Na.
```
Comment (vstinner) [inline Misc/NEWS.d/next/C API/2020-05-26-23-06-59.bpo-40778.7cumE-.rst:2]: I don't think that it deserves to be documented. It doesn't change the C API, no?

File: Modules/arraymodule.c [hunk 2]
```diff
@@ -2543,7 +2543,7 @@ array_buffer_getbuf(arrayobject *self, Py_buffer *view, int flags)
     view->suboffsets = NULL;
     view->shape = NULL;
     if ((flags & PyBUF_ND)==PyBUF_ND) {
-        view->shape = &((Py_SIZE(self)));
+        view->shape = (Py_ssize_t *)&((Py_SIZE(self)));
```
Comment (vstinner) [inline Modules/arraymodule.c]: Hum, I suggest:

```suggestion
        view->shape = &((PyVarObject *)self)->ob_size;
```

Comment (corona10) [reply Modules/arraymodule.c]: Good!

**Comment by vstinner** [comment]
I created PR #20429.

**Pull request closed without merging**